internal/resources/resource_network.go

```diff
@@ -420,3 +445,17 @@ func resourceMetalNetworkDelete(d *schema.ResourceData, meta interface{}) (err e
 
 	return p.RefreshAvailableResources()
 }
+
+// getSupportedNetworkPurpose returns a string containing supported network purpose values.
+func getSupportedNetworkPurpose() string {
+	return fmt.Sprintf("%v, %v, %v, %v, %v, %v, %v, %v or %v",
+		rest.NETWORKPURPOSE_BACKUP,
+		rest.NETWORKPURPOSE_STORAGE,
```

Comment:
Any way to get this from API declarations?  On Operator API, there are "static" handlers to return this types of valid values for some attribute (e.g. pod types, network types, network purposes).  I don't think we have equivalent on Client side.  Should at a minimum put comments on quake side that if this list changes, need to update this file as well.

Reply:
Looked into swagger/openapi Go SDK generation and does not look like there is a way to do this. Will add comments to the 'quake' repo.